### PR TITLE
[DEV-3197] Prevent graphql call from querying referrer when disabled

### PR DESCRIPTION
**Pull request by 00salmon**
## Description of the change

> Fixes issue for tenants that have accounts with multiple users

## Type of change

- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation or Development tools (readme, specs, tests, code formatting)

## Links

- Jira issue number: https://saasquatch.atlassian.net/browse/DEV-3197
- Process.st launch checklist: https://app.process.st/runs/DEV-3197-Prevent-vanilla-component-referral-table-from-querying-showReferrer-when-disabled-hG1Bv7qJ7cVpBZ8wOwtGMA

Testing links:
showReferrer="true"
[squathjs](https://squathjs-demo-qa.surge.sh/?sandbox=eyJzY3JpcHQiOiJodHRwczovL2Zhc3Quc3NxdC5pby9zcXVhdGNoLWpzQDIiLCJkb21haW4iOiJodHRwczovL3N0YWdpbmcucmVmZXJyYWxzYWFzcXVhdGNoLmNvbSIsInRlbmFudEFsaWFzIjoidGVzdF9hOGI0MWpvdGY4YTF2IiwiZGVidWciOnRydWUsImluaXRPYmoiOnsid2lkZ2V0VHlwZSI6InAvemN4dnpjdnh2emN4L3cvcmVmZXJyZXJXaWRnZXQiLCJlbmdhZ2VtZW50TWVkaXVtIjoiRU1CRUQiLCJ1c2VyIjp7ImlkIjoiamFjayIsImFjY291bnRJZCI6ImpvZSJ9LCJqd3QiOiJleUpoYkdjaU9pSklVekkxTmlJc0luUjVjQ0k2SWtwWFZDSjkuZXlKMWMyVnlJanA3SW1sa0lqb2lhbUZqYXlJc0ltRmpZMjkxYm5SSlpDSTZJbXB2WlNKOWZRLlo1VGNWSzFGeExsMGpQWlpCaGpaclc3TVowdENmVThiS0hwbmlqUkdBMGsifX0%3D)

showReferrer="false"
[squathjs](https://squathjs-demo-qa.surge.sh/?sandbox=eyJzY3JpcHQiOiJodHRwczovL2Zhc3Quc3NxdC5pby9zcXVhdGNoLWpzQDIiLCJkb21haW4iOiJodHRwczovL3N0YWdpbmcucmVmZXJyYWxzYWFzcXVhdGNoLmNvbSIsInRlbmFudEFsaWFzIjoidGVzdF9hOGI0MWpvdGY4YTF2IiwiZGVidWciOnRydWUsImluaXRPYmoiOnsid2lkZ2V0VHlwZSI6InAveWV0LWFub3RoZXIvdy9yZWZlcnJlcldpZGdldCIsImVuZ2FnZW1lbnRNZWRpdW0iOiJFTUJFRCIsInVzZXIiOnsiaWQiOiJqb2UiLCJhY2NvdW50SWQiOiJqb2UifSwiand0IjoiZXlKaGJHY2lPaUpJVXpJMU5pSXNJblI1Y0NJNklrcFhWQ0o5LmV5SjFjMlZ5SWpwN0ltRmpZMjkxYm5SSlpDSTZJbXB2WlNJc0ltbGtJam9pYW05bEluMTkuVnNWVUk5UXZXV0NXcmFrUFBwNGlHWGQxYVIwdDBSQkFaUUpkaF9zc3lNRSJ9fQ%3D%3D)

### Development

- [x] [Prettier](https://www.npmjs.com/package/prettier) was run (if applicable)
- [x] The behaviour changes in the pull request are covered by specs
- [x] All tests related to the changed code pass in development

### Paperwork

- [x] This pull request has a descriptive title and information useful to a reviewer
- [x] This pull request has a Jira number
- [x] This pull request has a Process.st launch checklist

### Code review

- [x] Changes have been reviewed by at least one other engineer
- [x] Security impacts of this change have been considered

### CODE REVIEW

**Comment by 00salmon** [comment]
note: vanilla-components has no changelog, needs one constructed

**Review by loganvolkers — APPROVED** [review]
Looks good to me

File: packages/vanilla-components/CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
+and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
+
+## [Unreleased]
+
+## [1.2.3] - 2022-03-01
+
+### Updated
+
+- \<sqh-referral-list> showReferrer prop now prevents referrer from being queried when disabled
```
Comment (00salmon) [inline packages/vanilla-components/CHANGELOG.md:14]:
```suggestion
- \<sqh-referral-list> showReferrer prop now prevents referrer from being queried when set to "false"
```

File: packages/vanilla-components/CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
+and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
+
+## [Unreleased]
+
+## [1.2.3] - 2022-03-01
+
+### Updated
+
+- \<sqh-referral-list> showReferrer prop now prevents referrer from being queried when set to "false"
```
Comment (00salmon) [inline packages/vanilla-components/CHANGELOG.md:14]:
```suggestion
- \<sqh-referral-list> show-referrer prop now prevents referrer from being queried when set to "false"
```

**Comment by DerekSiemens** [comment]
I know we don't really have specs for Vanilla components but I think this is a good opportunity to add some as we are going to do a small QA run anyways (like we always do). So I would like to see that here. Thank you for starting a changelog 🙏

File: packages/vanilla-components/CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,143 @@
+# Changelog
+
+All notable changes to this project will be documented in this file.
+
+The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
+and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
+
+## [Unreleased]
+
+## [1.2.3] - 2022-03-01
```
Comment (DerekSiemens) [inline packages/vanilla-components/CHANGELOG.md:10]: Flagging date update needed

Comment (DerekSiemens) [inline packages/vanilla-components/CHANGELOG.md:10]:
```suggestion
## [1.2.3] - 2022-03-02
```

File: packages/vanilla-components/CHANGELOG.md [hunk 4]
```diff
@@ -0,0 +1,143 @@
+# Changelog
+
+All notable changes to this project will be documented in this file.
+
+The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
+and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
+
+## [Unreleased]
+
+## [1.2.3] - 2022-03-02
```
Comment (00salmon) [inline packages/vanilla-components/CHANGELOG.md:10]:
```suggestion
## [1.2.3] - 2022-03-03
```

File: packages/vanilla-components/src/components/referral-list/referral-list.feature [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+  Scenario: Showing the referrer for a user with a duplicate accountId breaks the referral list
+    Given a user with the same accountId as another user
+    And they have been referred
+    But the referral list does not have the prop 'show-referrer="false"'
+    When the user views the referral list
+    Then the query fails with a permission error
+    And the referral table renders as "undefined"
+
+  @minutia
+  Scenario: Showing the referrer for a user
```
Comment (DerekSiemens) [inline packages/vanilla-components/src/components/referral-list/referral-list.feature:96]:
```suggestion
  Scenario: The referral list can show a users referrer
```

File: packages/vanilla-components/src/components/referral-list/referral-list.feature [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+    When I load my program widget
+    Then the user I referred in the referral list is shown
+    And the referral status for that user is "Reward Pending"
+    And the referral status is colored gray
+    When the referred user triggers the conversion
+    Then the referral status for that user says "Referred"
+    And the status is colored in green
+
+  @ui
+  Scenario Outline: Referral reward description
```
Comment (DerekSiemens) [inline packages/vanilla-components/src/components/referral-list/referral-list.feature:55]:
```suggestion
  Scenario Outline: The referral list displays reward descriptions based on the number of rewards and their status
```

File: packages/vanilla-components/src/components/referral-list/referral-list.feature [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+    Given I have a referral
+    And the number of rewards for that referral is <numberOfRewards>
```
Comment (DerekSiemens) [inline packages/vanilla-components/src/components/referral-list/referral-list.feature:57]: This spec is not proper, there are more variables in the examples then used in the specs - some steps are clearly missing. I do believe this is a historical spec, but might be worth fixing up quickly

File: packages/vanilla-components/CHANGELOG.md [hunk 8]
```diff
@@ -0,0 +1,143 @@
+# Changelog
+
+All notable changes to this project will be documented in this file.
+
+The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
+and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
+
+## [Unreleased]
+
+## [1.2.3] - 2022-03-03
```
Comment (DerekSiemens) [inline packages/vanilla-components/CHANGELOG.md:10]: Update me!

File: packages/vanilla-components/src/components/referral-list/referral-list.feature (hunk 7 above)
Comment (00salmon) [reply packages/vanilla-components/src/components/referral-list/referral-list.feature:57]: decided to remove the extra columns (expiryText, expiryDate). i was trying to make sense of them and some were just wrong / didn't make sense and others redundant based on the rewardDescription column

File: packages/vanilla-components/src/components/referral-list/referral-list.feature [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+      | AVAILABLE         |                   | 1               | does not      |
+      | PENDING           | Reward Pending    | 0               | doesn't have  |
+      | REDEEMED          | Redeemed          | 1               | doesn't have  |
+
+
+  @minutia
+  Scenario: Setting show-referrer prop to false prevents query for referredByReferral
+    Given a user with the same accountId as another user
+    And they have been referred
+    And the referral list has the prop 'show-referrer="false"'
```
Comment (DerekSiemens) [inline packages/vanilla-components/src/components/referral-list/referral-list.feature:81]:
```suggestion
    And the referral list has the prop 'showreferrer="false"'
```

File: packages/vanilla-components/src/components/referral-list/referral-list.feature [hunk 10]
```diff
@@ -0,0 +1,104 @@
...
+    When the user views the referral list
+    Then the query does not return data for referredByReferral
+    And the referral table renders successfully
+    But they do not see the referral that referred them
+
+  @landmine
+  Scenario: Showing the referrer for a user with a duplicate accountId breaks the referral list
+    Given a user with the same accountId as another user
+    And they have been referred
+    But the referral list does not have the prop 'show-referrer="false"'
```
Comment (DerekSiemens) [inline packages/vanilla-components/src/components/referral-list/referral-list.feature:91]:
```suggestion
    But the referral list does not have the prop 'showreferrer="false"'
```

File: packages/vanilla-components/src/components/referral-list/referral-list.feature [hunk 11]
```diff
@@ -0,0 +1,104 @@
...
+    When the user views the referral list
+    Then the query fails with a permission error
+    And the referral table renders as "undefined"
+
+  @minutia
+  Scenario: The referral list can show a users referrer
+    Given a user that has been referred
+    And they have been referred
+    And the referral list does not have the prop 'show-referrer="false"'
```
Comment (DerekSiemens) [inline packages/vanilla-components/src/components/referral-list/referral-list.feature:100]:
```suggestion
    And the referral list does not have the prop 'showreferrer="false"'
```

File: packages/vanilla-components/src/components/referral-list/referral-list.feature [hunk 12]
```diff
@@ -0,0 +1,104 @@
...
+    When the user views the referral list
+    Then the referral table renders successfully
+    And the referrer row is shown at the top of the list
```
Comment (DerekSiemens) [inline packages/vanilla-components/src/components/referral-list/referral-list.feature:103]: This is not correct

Comment (00salmon) [inline packages/vanilla-components/src/components/referral-list/referral-list.feature:103]:
```suggestion
    And the referrer row is shown as the last referral in the list
```

**Review by DerekSiemens — APPROVED** [review]
Thanks for adding a changelog Sam, historical specs could use some love but progress over perfection there - we have added and test specs for new behaviours so I am happy there

**Pull request merged**